File: gameplay/js/model/VertexLocation.js
var catan = catan || {};
catan.models = catan.models || {};

catan.models.VertexLocation = (function VertexLocationNameSpace () {

	var VertexLocation = (function VertexLocationClass() {

		/**
		VertexLocation Class 
		<pre>
		direction = NE | N | NW |SW | S | SE
		</pre>
		@class VertexLocation 
		@constructor
		@param {String} direction
		@param {int} x
		@param {int} y
		*/
		function VertexLocation(direction, x, y) {
			this.direction = direction;
			this.x = x;
			this.y = y;
		}

		/**
		GetX method
		<pre>
		</pre>
		@method getX
		@return {int} x
		*/
		VertexLocation.prototype.getX = function() {
		    return this.x;
		}

		/**
		GetY method
		<pre>
		</pre>
		@method getY
		@return {int} y
		*/
		VertexLocation.prototype.getY = function() {
		    return this.y;
		}

		/**
		GetDirection method
		<pre>
		</pre>
		@method getDirection
		@return {String} direction
		*/
		VertexLocation.prototype.getDirection = function() {
		    return this.direction;
		}


		return VertexLocation;
	})();


	return VertexLocation;
})();

File: server/src/catan/model/VertexLocation.java
package catan.model;

public class VertexLocation {

	private int x;
	private int y;
	private String direction;
	private transient final String[] VERTEX_STRINGS = new String[] {"W","NW","NE","E","SE","SW"};
	
	public VertexLocation(int x, int y, int direction) {
		
		this.x = x;
		this.y = y;
		this.direction = VERTEX_STRINGS[direction];
	}
	
	public VertexLocation(int x, int y, String direction) {
		
		this.x = x;
		this.y = y;
		this.direction = direction;
	}

	public int getX() {
		return x;
	}

	public void setX(int x) {
		this.x = x;
	}

	public int getY() {
		return y;
	}

	public void setY(int y) {
		this.y = y;
	}

	public String getDirection() {
		return direction;
	}
	
	public int getDirectionIndex() {
		
		for (int i = 0; i < VERTEX_STRINGS.length; i++)
			if (VERTEX_STRINGS[i].equals(direction))
				return i;
		return -1;
	}

	public void setDirection(String direction) {
		this.direction = direction;
	}
}
